server/queues/tasks/ImportTask.ts

```diff
@@ -161,15 +161,27 @@ export default abstract class ImportTask extends BaseTask<Props> {
   }
 
   /**
-   * Fetch the remote data needed for the import, by default this will download
-   * any file associated with the FileOperation, save it to a temporary file,
-   * and return the path.
+   * Fetch the remote data associated with the file operation as a Buffer.
    *
    * @param fileOperation The FileOperation to fetch data for
-   * @returns string
+   * @returns A promise that resolves to the data as a buffer.
    */
-  protected async fetchData(fileOperation: FileOperation) {
-    return fileOperation.buffer;
+  protected async fetchData(fileOperation: FileOperation): Promise<Buffer> {
```

Comment:
Unfortunately `JSZip` cannot be initialized with a stream, so import remains bottlenecked by loading the zip file.